search/includes/classes/queue/class-schema.php

```diff
@@ -0,0 +1,154 @@
+		add_action( 'wp_install', array( $this, 'create_table_during_install' ) );
+		add_action( 'wpmu_new_blog', array( $this, 'create_tables_during_multisite_install' ) );
+
+		// Remove table when a multisite subsite is deleted.
+		add_filter( 'wpmu_drop_tables', array( $this, 'remove_multisite_table' ) );
+
+		if ( ! $this->is_installed() ) {
+			// In limited circumstances, try creating the table.
+			add_action( 'shutdown', array( $this, 'maybe_create_table_on_shutdown' ) );
+		}
+	}
+
+	public function is_installed() {
+		$db_version = (int) get_transient( self::DB_VERSION_TRANSIENT );
+
+		return version_compare( $db_version, 0, '>' );
+	}
+
+	/**
+	 * Create table during initial install
+	 */
+	public function create_table_during_install() {
+		if ( 'wp_install' !== current_action() ) {
+			return;
+		}
+
+		$this->_prepare_table();
+	}
+
+	/**
+	 * Create table when new subsite is added to a multisite
+	 *
+	 * @param int $bid Blog ID.
+	 */
+	public function create_tables_during_multisite_install( $bid ) {
+		switch_to_blog( $bid );
+
+		if ( ! self::is_installed() ) {
+			$this->_prepare_table();
+		}
+
+		restore_current_blog();
+	}
+
+	/**
+	 * When deleting a subsite from a multisite instance, include the plugin's table
+	 *
+	 * Core only drops its tables
+	 *
+	 * @param  array $tables_to_drop Array of prefixed table names to drop.
+	 * @return array
+	 */
+	public function remove_multisite_table( $tables_to_drop ) {
+		$tables_to_drop[] = $this->get_table_name();
+
+		return $tables_to_drop;
+	}
+
+	/**
+	 * For certain requests, create the table on shutdown
+	 * Does not include front-end requests
+	 */
+	public function maybe_create_table_on_shutdown() {
+		if ( ! is_admin() && ! ( defined( 'WP_CLI' ) && WP_CLI ) ) {
+			return;
+		}
+
+		$this->prepare_table();
+	}
+
+	/**
+	 * Create table in non-setup contexts, with some protections
+	 */
+	public function prepare_table() {
+		// Table installed and current version
+		if ( self::is_installed() ) {
+			return;
+		}
+
+		// Limit chance of race conditions when creating table.
+		$create_lock_set = wp_cache_add( self::TABLE_CREATE_LOCK, 1, null, 1 * \MINUTE_IN_SECONDS );
+
+		if ( false === $create_lock_set ) {
+			return;
+		}
+
+		$this->_prepare_table();
+	}
+
+	/**
+	 * Create the plugin's DB table when necessary
+	 */
+	protected function _prepare_table() {
+		// Use Core's method of creating/updating tables.
+		if ( ! function_exists( 'dbDelta' ) ) {
+			require_once ABSPATH . '/wp-admin/includes/upgrade.php';
+		}
+
+		global $wpdb;
+
+		$table_name = $this->get_table_name();
+
+		// Define schema and create the table.
+		$schema = "CREATE TABLE `{$table_name}` (
+			`job_id` bigint(20) NOT NULL AUTO_INCREMENT,
+			`object_id` bigint(20) DEFAULT NULL COMMENT 'WP object ID',
+			`object_type` varchar(45) DEFAULT NULL COMMENT 'WP object type',
+			`priority` int(1) DEFAULT '5' COMMENT 'Relative priority for this item compared to others (of any object_type)',
```

Comment:
Oh, TIR(emembered): https://stackoverflow.com/questions/12839927/mysql-tinyint-2-vs-tinyint1-what-is-the-difference

`(1)` just indicates display width for int variant columns. In this case, for optimization, we probably just want `tinyint` instead.

```suggestion
			`priority` tinyint(1) DEFAULT '5' COMMENT 'Relative priority for this item compared to others (of any object_type)',
```